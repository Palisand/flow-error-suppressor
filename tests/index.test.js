const {spawnSync} = require("child_process");
const {resolve} = require("path");

const buildSnapshotTest = (testdir) => {
  test("matches snapshot", () => {
    const {stdout: flowStdout} = spawnSync(
      "script",
      [
        "-q",
        "/dev/null",
        "flow",
        "--show-all-errors",
      ],
      {
        cwd: resolve(__dirname, testdir),
        encoding: "utf8"
      }
    );
    const {status, stderr, stdout} = spawnSync(
      "node",
      ["index.js"],
      {
        encoding: "utf8",
        input: flowStdout
      }
    );
    expect(stderr).toMatchSnapshot();
    expect(stdout).toMatchSnapshot();
    expect(status).toBe(0);
  })
};

[
  "no-errors",
  "package-errors",
  "script-errors"
].map(
  testdir => {
    describe(testdir, () => {
      buildSnapshotTest(testdir)
    })
  }
);
