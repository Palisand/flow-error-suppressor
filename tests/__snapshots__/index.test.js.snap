// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no-errors matches snapshot 1`] = `""`;

exports[`no-errors matches snapshot 2`] = `
"^DNo errors!
      0 suppressed
      0 shown

"
`;

exports[`package-errors matches snapshot 1`] = `""`;

exports[`package-errors matches snapshot 2`] = `
"
Found 1 error
      1 suppressed
      0 shown

"
`;

exports[`script-errors matches snapshot 1`] = `""`;

exports[`script-errors matches snapshot 2`] = `
"^D[31;1mError:[0m[0m [0m[0;4mindex.js:4[0m[0m
[0m[0;1m  4: [0m[0mfoo([0m[31m42[0m[0m);[0m[0;1m
         ^^ number. This type is incompatible with the expected param type of[0m[0m[0m[0m
[0m[0;1m  3: [0m[0mfunction foo(bar: [0m[31mstring[0m[0m) {}[0m[0;1m
                       ^^^^^^ string[0m[0m[0m[0m
[0m[0m
[0m
Found 1 error
      0 suppressed
      1 shown

"
`;
